priv/app/graph_panel.jsx

```diff
@@ -50,24 +52,31 @@ export default class GraphPanel extends React.Component {
   }
 
   handleFuns(data) {
-    this.state.funs = data;
-    this.setState(this.state);
+    if (this.state.funs.length !== data.length) {
```

Comment:
another edge case when the length is the same but the content is not. This can happen if one function is stopped and another added within poll interval of 500ms by another GUI. A human is probably not this fast but with the upcoming feature of editing fun in place this fast swap is possible.
I think `__.isEqual(this.state.funs, data)` does a deep comparison.